source/beamline/PGM/PGMBPMControl.cpp

```diff
@@ -0,0 +1,38 @@
+#include "beamline/PGM/PGMBPMControl.h"
+
+PGMBPMControl::PGMBPMControl(const QString &name, const QString &pvName, int avgValue, int variance, QObject *parent)
+    : QObject(parent)
+
+{
+    currentValue_ = 0;
+    averageValue_ = avgValue;
+    variance_ = variance;
+    state_ = true;
+    bpmPV_ = new AMReadOnlyPVControl(name, pvName, this);
+
+    // Get updated pv value from AMReadOnlyPVControl.
+    connect(bpmPV_, SIGNAL(valueChanged(double)), this, SLOT(setCurrentValue(double)));
+
+}
+
+void PGMBPMControl::updateValidity(double newValue){
+
+    int upperBound = averageValue_ + variance_;
+    int lowerBound = averageValue_ - variance_;
+
+    bool newState = (newValue < upperBound || newValue > lowerBound);
+
+    if (newState != state_){
+
+        state_ = newState;
+        emit stateValidityChanged(state_);
+    }
+}
+
+void PGMBPMControl::setCurrentValue(double value){
+
+    currentValue_ = value;
+    updateValidity(currentValue_);
+    emit onBPMValueChanged(QString("%1 um").arg(value, 0, 'f', 0));
```

Comment:
Oh, shoot.  Sorry @dleacock.  I think I misunderstood your original signal so you'll need to make one more change. 

In general, there is no need for a control to emit a value changed signal as a string like this (there are some exceptions, but this is not one of them).  The model should emit the value as a double (which is the real data) and then the view would have the `QString("%1 um").arg(value, 0, 'f', 0)` call inside it's `onBPMValueChanged(double)` slot.  The main reason for this is you make it harder for models to talk to each other by having your model return a string as the value.